frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/WorkLoadManager.java

```diff
@@ -86,19 +89,28 @@ public boolean hasNoWorker(String modelName) {
         if (currentWorkers < minWorker) {
             return addThreads(threads, model, minWorker - currentWorkers);
         } else {
-            for (int i = currentWorkers - 1; i >= minWorker; --i) {
-                WorkerThread thread = threads.remove(i);
-                thread.shutdown();
+            for (int i = 0; i < currentWorkers - 1; i++) {
+                if (i >= minWorker) {
+                    WorkerThread thread = threads.remove(i);
+                    thread.shutdown();
+                } else {
+                    if (threads.get(i).getState() == WorkerState.WORKER_ERROR
+                            || threads.get(i).getState() == WorkerState.WORKER_STOPPED) {
+                        status = APIStatus.PARTIAL_SUCCESS;
```

Comment:
Should PARTIAL_SUCCESS only apply to scale up?

Reply:
I think this should apply to any scale call. For example, this should also apply for two consecutive calls for same number of workers.

Reply:
This behavior is not consistent with scale up.
When scale down, if any existing worker is failed, we return partial success, but scale up, we only check added worker.

The simplest behavior is return partial if:
1. Success = all worker are healthy
2. At least worker is success, but have some worker failed
3. Failed = all worker are failed

We actually don't need any of these changes. We can simply loop all workers to check health status before API return.

Reply:
True, its simpler to hold all the changes in the same method.